src/Mod/Spreadsheet/App/Cell.cpp

```diff
@@ -301,40 +302,80 @@ void Cell::setContent(const char * value)
-                    if (expr)
-                        delete expr->eval();
-                }
-                catch (Base::Exception &) {
-                    expr = new App::StringExpression(owner->sheet(), value);
+                    auto parsedExpr = App::ExpressionParser::parse(owner->sheet(), value);
+
+                    if (const auto fraction = freecad_dynamic_cast<OperatorExpression>(parsedExpr)) {
+                        if (fraction->getOperator() == OperatorExpression::UNIT) {
+                            const auto left = freecad_dynamic_cast<NumberExpression>(fraction->getLeft());
+                            const auto right = freecad_dynamic_cast<UnitExpression>(fraction->getRight());
+                            if (left && right) {
+                                expr = parsedExpr;
+                            }
+                        }
+                        else if (fraction->getOperator() == OperatorExpression::DIV) {
+                            // only the following types of fractions are ok:
+                            //     1/2, 1m/2, 1/2s, 1m/2s, 1/m
+
+                            // check for numbers in (de)nominator
+                            const bool isNumberNom = freecad_dynamic_cast<NumberExpression>(fraction->getLeft());
+                            const bool isNumberDenom = freecad_dynamic_cast<NumberExpression>(fraction->getRight());
+
+                            // check for numbers with units in (de)nominator
+                            const auto opNom = freecad_dynamic_cast<OperatorExpression>(fraction->getLeft());
+                            const auto opDenom = freecad_dynamic_cast<OperatorExpression>(fraction->getRight());
+                            const bool isQuantityNom = opNom && opNom->getOperator() == OperatorExpression::UNIT;
+                            const bool isQuantityDenom = opDenom && opDenom->getOperator() == OperatorExpression::UNIT;
+
+                            // check for units in denomainator
+                            const auto uDenom = freecad_dynamic_cast<UnitExpression>(fraction->getRight());
+                            const bool isUnitDenom = uDenom && uDenom->getTypeId() == UnitExpression::getClassTypeId();
+
+                            const bool isNomValid = isNumberNom || isQuantityNom;
+                            const bool isDenomValid = isNumberDenom || isQuantityDenom || isUnitDenom;
+                            if (isNomValid && isDenomValid) {
+                                expr = parsedExpr;
+                            }
+                        }
+                    }
+                    // this should never be reached since numberExpressions should already have been caught with strtod
+                    else if (const auto number = freecad_dynamic_cast<NumberExpression>(parsedExpr)) {
+                        Console().Error("This should never reach this point\n");
```

Comment:
If it does ever get hit, that error is going to look pretty obscure on the console. If you're going to bother having it, I'd consider writing an error message that at least give the reader a hint at where to look, e.g.
```suggestion
                        Console().Error("Spreadsheet should already have parsed this as a number and didn't.\n");
```